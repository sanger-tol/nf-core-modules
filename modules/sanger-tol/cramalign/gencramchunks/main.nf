process CRAMALIGN_GENCRAMCHUNKS {
    label "process_single"
    executor "local"

    input:
    // Native processes can't take path values as inputs
    tuple val(meta), val(crai)
    val cram_bin_size

    output:
    tuple val(meta), val(chunkn), val(slices), emit: cram_slices
    path("versions.yml")                     , emit: versions

    when:
    task.ext.when == null || task.ext.when

    exec:
    // Note: Manually bump version number when updating module
    def VERSION = "1.0.1"

    def n_slices = file(crai).countLines(decompress: true) - 1
    def size     = cram_bin_size
    def n_bins   = n_slices.intdiv(size)
    chunkn       = (0..n_bins).collect()
    slices       = chunkn.collect { chunk ->
        def lower = chunk * size
        def upper = [lower + size - 1, n_slices].min()

        return [ lower, upper ]
    }
    def versions_file = new File("${task.workDir}/versions.yml")
    versions_file.write(
        """
        CRAMALIGN_GENCRAMCHUNKS:
            cramalign_gencramchunks: ${VERSION}
        """
    )
}
