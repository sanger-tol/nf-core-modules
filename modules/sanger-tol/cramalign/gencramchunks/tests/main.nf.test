nextflow_process {

    name "Test Process CRAMALIGN_GENCRAMCHUNKS"
    script "../main.nf"
    process "CRAMALIGN_GENCRAMCHUNKS"

    tag "modules"
    tag "modules_sangertol"
    tag "cramalign"
    tag "cramalign/gencramchunks"

    test("cram") {

        when {
            process {
                """
                input[0] = [
                    [ id:'test' ], // meta map
                    file(params.modules_testdata_base_path + 'Undibacterium_unclassified/genomic_data/baUndUnlc1/hic-arima2/41741_2%237.sub.cram.crai', checkIfExists: true)
                ]
                input[1] = 10
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
