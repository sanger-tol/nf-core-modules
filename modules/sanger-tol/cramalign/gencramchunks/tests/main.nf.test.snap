{
    "cram": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        [
                            0,
                            1,
                            2,
                            3,
                            4,
                            5,
                            6,
                            7,
                            8,
                            9,
                            10,
                            11,
                            12,
                            13,
                            14,
                            15,
                            16,
                            17,
                            18,
                            19,
                            20,
                            21,
                            22,
                            23,
                            24,
                            25,
                            26,
                            27,
                            28,
                            29,
                            30,
                            31,
                            32,
                            33,
                            34,
                            35,
                            36,
                            37,
                            38
                        ],
                        [
                            [
                                0,
                                9
                            ],
                            [
                                10,
                                19
                            ],
                            [
                                20,
                                29
                            ],
                            [
                                30,
                                39
                            ],
                            [
                                40,
                                49
                            ],
                            [
                                50,
                                59
                            ],
                            [
                                60,
                                69
                            ],
                            [
                                70,
                                79
                            ],
                            [
                                80,
                                89
                            ],
                            [
                                90,
                                99
                            ],
                            [
                                100,
                                109
                            ],
                            [
                                110,
                                119
                            ],
                            [
                                120,
                                129
                            ],
                            [
                                130,
                                139
                            ],
                            [
                                140,
                                149
                            ],
                            [
                                150,
                                159
                            ],
                            [
                                160,
                                169
                            ],
                            [
                                170,
                                179
                            ],
                            [
                                180,
                                189
                            ],
                            [
                                190,
                                199
                            ],
                            [
                                200,
                                209
                            ],
                            [
                                210,
                                219
                            ],
                            [
                                220,
                                229
                            ],
                            [
                                230,
                                239
                            ],
                            [
                                240,
                                249
                            ],
                            [
                                250,
                                259
                            ],
                            [
                                260,
                                269
                            ],
                            [
                                270,
                                279
                            ],
                            [
                                280,
                                289
                            ],
                            [
                                290,
                                299
                            ],
                            [
                                300,
                                309
                            ],
                            [
                                310,
                                319
                            ],
                            [
                                320,
                                329
                            ],
                            [
                                330,
                                339
                            ],
                            [
                                340,
                                349
                            ],
                            [
                                350,
                                359
                            ],
                            [
                                360,
                                369
                            ],
                            [
                                370,
                                379
                            ],
                            [
                                380,
                                380
                            ]
                        ]
                    ]
                ],
                "1": [
                    "versions.yml:md5,937b65ae8a1a62094e422deed4aa2480"
                ],
                "cram_slices": [
                    [
                        {
                            "id": "test"
                        },
                        [
                            0,
                            1,
                            2,
                            3,
                            4,
                            5,
                            6,
                            7,
                            8,
                            9,
                            10,
                            11,
                            12,
                            13,
                            14,
                            15,
                            16,
                            17,
                            18,
                            19,
                            20,
                            21,
                            22,
                            23,
                            24,
                            25,
                            26,
                            27,
                            28,
                            29,
                            30,
                            31,
                            32,
                            33,
                            34,
                            35,
                            36,
                            37,
                            38
                        ],
                        [
                            [
                                0,
                                9
                            ],
                            [
                                10,
                                19
                            ],
                            [
                                20,
                                29
                            ],
                            [
                                30,
                                39
                            ],
                            [
                                40,
                                49
                            ],
                            [
                                50,
                                59
                            ],
                            [
                                60,
                                69
                            ],
                            [
                                70,
                                79
                            ],
                            [
                                80,
                                89
                            ],
                            [
                                90,
                                99
                            ],
                            [
                                100,
                                109
                            ],
                            [
                                110,
                                119
                            ],
                            [
                                120,
                                129
                            ],
                            [
                                130,
                                139
                            ],
                            [
                                140,
                                149
                            ],
                            [
                                150,
                                159
                            ],
                            [
                                160,
                                169
                            ],
                            [
                                170,
                                179
                            ],
                            [
                                180,
                                189
                            ],
                            [
                                190,
                                199
                            ],
                            [
                                200,
                                209
                            ],
                            [
                                210,
                                219
                            ],
                            [
                                220,
                                229
                            ],
                            [
                                230,
                                239
                            ],
                            [
                                240,
                                249
                            ],
                            [
                                250,
                                259
                            ],
                            [
                                260,
                                269
                            ],
                            [
                                270,
                                279
                            ],
                            [
                                280,
                                289
                            ],
                            [
                                290,
                                299
                            ],
                            [
                                300,
                                309
                            ],
                            [
                                310,
                                319
                            ],
                            [
                                320,
                                329
                            ],
                            [
                                330,
                                339
                            ],
                            [
                                340,
                                349
                            ],
                            [
                                350,
                                359
                            ],
                            [
                                360,
                                369
                            ],
                            [
                                370,
                                379
                            ],
                            [
                                380,
                                380
                            ]
                        ]
                    ]
                ],
                "versions": [
                    "versions.yml:md5,937b65ae8a1a62094e422deed4aa2480"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.2"
        },
        "timestamp": "2025-10-03T13:14:09.683352"
    }
}